src/QtUtils/ExecuteProcess.cpp

```diff
@@ -0,0 +1,110 @@
+// Copyright (c) 2021 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "QtUtils/ExecuteProcess.h"
+
+#include <absl/strings/str_format.h>
+#include <absl/time/time.h>
+
+#include <QMetaEnum>
+#include <QProcess>
+#include <QTimer>
+#include <memory>
+
+#include "OrbitBase/Future.h"
+#include "OrbitBase/Promise.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_qt_utils {
+
+using orbit_base::Future;
+
+Future<ErrorMessageOr<QByteArray>> ExecuteProcess(const QString& program,
+                                                  const QStringList& arguments, QObject* parent,
+                                                  absl::Duration timeout) {
+  CHECK(parent != nullptr);
+
+  auto promise = std::make_shared<orbit_base::Promise<ErrorMessageOr<QByteArray>>>();
+
+  // Create and connect QProcess
+  QProcess* process = new QProcess(parent);
+  process->setProgram(program);
+  process->setArguments(arguments);
+
+  std::string process_description =
+      arguments.empty()
+          ? program.toStdString()
+          : absl::StrFormat("%s %s", program.toStdString(), arguments.join(" ").toStdString());
+
+  QObject::connect(
+      process, &QProcess::errorOccurred, parent,
+      [promise, process, process_description](QProcess::ProcessError error) {
+        std::string error_message = absl::StrFormat(
+            "Error occurred while executing process \"%s\", error: %s, stdout:\n%s\nstderr:\n%s\n",
```

Comment:
```suggestion
            "Error occurred while executing process \"%s\", error: %s,\nstdout:\n%s\nstderr:\n%s\n",
```
You do this below, maybe you want to do the same here, or viceversa, for what it's worth.